src/spirv.cc

```diff
@@ -737,6 +745,14 @@ class SPIRVmodule {
             SpillAnnotation.push_back(std::make_pair(ArgIndex, ArgSize));
           }
         }
+
+        // A magic variable created by HipIGBADetector.cpp.
+        if (Name == "__chip_module_may_have_IGBAs") {
```

Comment:
Would it be safer to reverse the logic of the flag (also for backwards compatibility)? That is, add the magic variable only if known that there are no IGBAs in the module.

Reply:
SGTM. Updated.